utilities/transactions/lock/range/range_tree/range_tree_lock_tracker.h

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+// This source code is licensed under both the GPLv2 (found in the
+// COPYING file in the root directory) and Apache 2.0 License
+// (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "util/mutexlock.h"
+#include "utilities/transactions/lock/lock_tracker.h"
+
+// Range Locking:
+#include "lib/locktree/lock_request.h"
+#include "lib/locktree/locktree.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+class RangeLockList;
+
+/*
+  Storage for locks that are currently held by a transaction.
+
+  Locks are kept in toku::range_buffer because toku::locktree::release_locks()
+  accepts that as an argument.
+
+  Note: the list of locks may differ slighly from the contents of the lock
+  tree, due to concurrency between lock acquisition, lock release, and lock
+  escalation. See MDEV-18227 and RangeTreeLockManager::UnLock for details.
+  This property is currently harmless.
+*/
+class RangeLockList /*: public PessimisticTransaction::Lock--Storage */ {
+ public:
+  virtual ~RangeLockList() { clear(); }
+
+  void clear() {
+    for (auto it : buffers_) {
+      it.second->destroy();
+    }
+    buffers_.clear();
+  }
+
+  RangeLockList() : releasing_locks_(false) {}
+
+  void append(uint32_t cf_id, const DBT* left_key, const DBT* right_key) {
+    MutexLock l(&mutex_);
+    // there's only one thread that calls this function.
+    // the same thread will do lock release.
+    assert(!releasing_locks_);
+    auto it = buffers_.find(cf_id);
+    if (it == buffers_.end()) {
+      // create a new one
+      it = buffers_
+               .emplace(cf_id, std::shared_ptr<toku::range_buffer>(
+                                   new toku::range_buffer()))
+               .first;
+      it->second->create();
+    }
+    it->second->append(left_key, right_key);
+  }
+
+  std::unordered_map<uint32_t, std::shared_ptr<toku::range_buffer>> buffers_;
```

Comment:
why are these member variables public?

Reply:
Ok, moved other code around and that allowed to make them private.